### PR TITLE
ocamlPackages.ocaml-lsp: 1.5.0 -> 1.7.0

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change

update

tested on a project both with ocaml 4.07 and 4.12

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/pp/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+, fetchFromGitHub
+, ppx_expect
+, lib
+}:
+
+buildDunePackage rec {
+  pname = "pp";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/pp/default.nix:11]: You can use the tarball from `https://github.com/ocaml-dune/pp/releases/download/1.1.2/pp-1.1.2.tbz` as source; it includes correct version information.

File: pkgs/development/ocaml-modules/pp/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    owner = "ocaml-dune";
+    repo = "pp";
+    rev = version;
+    sha256 = "0w7gxa85ffbd6jgs6ziarq69yi423f0qkpk05r3abh6lg8smw8pg";
+  };
+
+  useDune2 = true;
+  minimumOCamlVersion = "4.08";
+
+  checkInputs = [ ppx_expect ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/pp/default.nix:21]:
```suggestion
  checkInputs = [ ppx_expect ];
  doCheck = true;
```

File: pkgs/development/ocaml-modules/pp/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ocaml-dune";
+    repo = "pp";
+    rev = version;
+    sha256 = "0w7gxa85ffbd6jgs6ziarq69yi423f0qkpk05r3abh6lg8smw8pg";
+  };
+
+  useDune2 = true;
+  minimumOCamlVersion = "4.08";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/pp/default.nix:19]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/pp/default.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/development/ocaml-modules/pp/default.nix]: done

File: pkgs/development/ocaml-modules/pp/default.nix (hunk 3 above)
Comment (symphorien) [reply pkgs/development/ocaml-modules/pp/default.nix]: oops, thanks

File: pkgs/development/ocaml-modules/pp/default.nix (hunk 2 above)
Comment (symphorien) [reply pkgs/development/ocaml-modules/pp/default.nix:19]: done

File: pkgs/development/ocaml-modules/pp/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+, ppx_expect
+, lib
+}:
+
+buildDunePackage rec {
+  pname = "pp";
+  version = "1.1.2";
+
+  src = fetchzip {
+    url = "https://github.com/ocaml-dune/pp/releases/download/${version}/pp-${version}.tbz";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/pp/default.nix:12]: Why don’t you use `fetchurl`?

Comment (symphorien) [reply pkgs/development/ocaml-modules/pp/default.nix:12]: I use those quite interchangeably, can you tell me where/when I should use one and not the other ?

**Pull request merged**